# Checking The Alignment Quality

## Coverage
The coverage can be calculated by [GATK](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.php) (Only available in GATK 3, see [here](https://gatkforums.broadinstitute.org/gatk/discussion/11535/depth-of-coverage-in-gatk-4#latest) absent in GATK 4), [Samtools](https://www.biostars.org/p/5165/), [BEDTools](https://bedtools.readthedocs.io/en/latest/content/tools/genomecov.html), ILLUMINA [Calculator](https://support.illumina.com/downloads/sequencing_coverage_calculator.html).

Read [here](https://www.ecseq.com/support/ngs/how-to-calculate-the-coverage-for-a-sequencing-experiment) to understand what is Coverage in different context.

How to deal with multimapping reads: refer to this [post](https://gist.github.com/sbamin/27acf13f2a28161efbf89a273559bca4)

